blog/views.py

```diff
@@ -17,15 +21,30 @@
 #     # queryset = Entry.objects.published()
 #     template_name = "home.html"
 #     paginate_by = 2
-class EntryView(ListView):
-    template_name = 'Entry/Entry.html'
+class EntryView(DetailView):
+    template_name = 'blog/view_entry.html'
+    model = Entry
+    pk_url_kwarg = 'entry_id'
+
+    def get_context_data(self, **kwargs):
+        context = super(EntryView, self).get_context_data(**kwargs)
+
+        return context
+
+class EntryDeleteView(DeleteView):
+    template_name = 'blog/delete_entry.html'
+    model = Entry
+
+
+class EntriesView(ListView):
+    template_name = 'blog/blog.html'
     model = Entry
 
 
 def index(request):
```

Comment:
These function based views should be removed and replaced with class based views.